lib/decoders/kmz.js

```diff
@@ -49,6 +48,32 @@ class KMZ extends Duplex {
   }
 
   _onBuffered() {
+    this.emit('readable');
+  }
+
+  _onOpenKmlStream(kmlStream, zipFile) {
+    kmlStream
+      .pipe(this._kmlDecoder)
+      .on('error', (err) => {
+        this.emit('error', err);
+      })
+      .on('data', (data) => {
```

Comment:
Seems like a backpressure mechanism? If we fail to do the expected thing (push the data), wait for things to clear up before continuing to process.